resources/gui/js/api.js

```diff
@@ -0,0 +1,543 @@
+      try {
+        let instance;
+
+        if (typeof Api === 'string' && String(Api).slice(-3) === 'Api') {
+          // oof
+          instance = eval(`new ${Api}()`);
+        } else {
+          instance = new Api();
+        }
+
+        this.register(instance.name, instance);
+        instance.init();
+      } catch (e) {
+        console.error(`Could not initialize ${Api}`);
+      }
+    });
+  }
+
+  /**
+   * Initialize third party drop downs,
+   * add input event listener,
+   * initialize tooltips
+   */
+  static initInputs() {
+    this.initDropDowns();
+    this.initChecklabelInputs();
+    this.initRadiolabelInputs();
+    this.initTooltips();
+    this.initDataCalls();
+  }
+
+  /**
+   * @see {initInputs}
+   * TODO Remove jQuery
+   */
+  static initDropDowns() {
+    const dropdowns = $('.simple-value-dropdown');
+    dropdowns.selectpicker();
+
+    dropdowns.on('change', function () {
+      if (this.id !== '') {
+        CosmoScout.callNative(this.id, this.value);
+      }
+    });
+
+    return;
+
+    document.querySelectorAll('.simple-value-dropdown').forEach((dropdown) => {
+      if (typeof dropdown.selectpicker !== 'undefined') {
+        dropdown.selectpicker();
+      }
+
+      if (typeof dropdown.dataset.initialized !== 'undefined') {
+        return;
+      }
+
+      dropdown.addEventListener('change', (event) => {
+        if (event.target !== null && event.target.id !== '') {
+          CosmoScout.callNative(event.target.id, event.target.value);
+        }
+      });
+
+      dropdown.dataset.initialized = 'true';
+    });
+  }
+
+  /**
+   * @see {initInputs}
+   */
+  static initChecklabelInputs() {
+    document.querySelectorAll('.checklabel input').forEach((input) => {
+      if (typeof input.dataset.initialized !== 'undefined') {
+        return;
+      }
+
+      input.addEventListener('change', (event) => {
+        if (event.target !== null) {
+          CosmoScout.callNative(event.target.id, event.target.checked);
+        }
+      });
+
+      input.dataset.initialized = 'true';
+    });
+  }
+
+  /**
+   * @see {initInputs}
+   */
+  static initRadiolabelInputs() {
+    document.querySelectorAll('.radiolabel input').forEach((input) => {
+      if (typeof input.dataset.initialized !== 'undefined') {
+        return;
+      }
+
+      input.addEventListener('change', (event) => {
+        if (event.target !== null) {
+          CosmoScout.callNative(event.target.id);
+        }
+      });
+
+      input.dataset.initialized = 'true';
+    });
+  }
+
+  /**
+   * @see {initInputs}
+   * @see {callNative}
+   * Adds an onclick listener to every element containing [data-call="methodname"]
+   * The method name gets passed to CosmoScout.callNative.
+   * Arguments can be passed by separating the content with ','
+   * E.g.: fly_to,Africa -> CosmoScout.callNative('fly_to', 'Africa')
+   *       method,arg1,...,argN -> CosmoScout.callNative('method', arg1, ..., argN)
+   */
+  static initDataCalls() {
```

Comment:
I think this method can be improved somehow. Here are some issues I see:
* Does it work for multiple number arguments?
* What about boolean arguments?
* Last but not least, `data-call="set_quality,5f"` looks a bit ugly

Have you considered just to use `eval()`? It should be possible to write `data-call="CosmoScout.callNative('set_quality', 5)"` and on click you just call `eval(input.dataset.call)`. What do you think?

Reply:
This is now resolved by using eval. The contents of the data-call attribute will be passed to `CosmoScout.callNative`.  

Example:
```
<a data-call="'set_quality',5"> => CosmoScout.callNative('set_quality', 5)
```
Notice `'` around `set_quality`. This is needed to denote strings, as `eval` wouldn't parse it as one otherwise.